mavis/assemble.py

```diff
@@ -107,12 +107,12 @@ def trim_tails_by_freq(self, min_weight):
                     if data['freq'] < min_weight:
                         self.remove_edge(src, tgt)
                     if src not in visited:
-                        ends.add(src)
+                        ends.append(src)
                     if tgt not in visited:
-                        ends.add(tgt)
+                        ends.append(tgt)
 
         # remove any resulting singlets
-        for node in visited:
+        for node in sorted([n for n in visited if self.degree(n) == 0], reverse=True):
```

Comment:
Sorting this won't impact the results since these are singlets

Reply:
okay, will remove this one.